### PR TITLE
(GH-354) Document PDK as source information

### LINKED ISSUE #354: Update readme to explain Puppet-Agent default
Since we decided to make the PDK-as-source a feature toggle, the readme needs to reflect the current default of Puppet-Agent, but also instruct how to use the new feature

### PR DESCRIPTION
**Pull request by jpogran**
This commit updates the readme to explain how to use either the PDK or Puppet-Agent as a source for the extension. It also documents the new configuration options to allow this.

Fixes #354

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,7 +65,17 @@ This extension provides full Puppet Language support for [Visual Studio Code](ht
 
 ## Requirements
 
-You will need to have Puppet Agent installed in order to fully use this extension. You can find instructions and installation links here:
+You will need to have the [Puppet Development Kit (PDK)](https://puppet.com/docs/pdk/1.x/pdk.html) or [Puppet Agent](https://puppet.com/docs/puppet/4.10/about_agent.html) installed in order to fully use this extension.
+
+**Note:** PDK version 1.5.0 or higher is required, as that was the first version to include the Puppet gem.
```
Comment (glennsarti) [inline README.md]: I don't think we need to go into the specifics.  Perhaps `You will need to have the [Puppet Development Kit (PDK)](https://puppet.com/docs/pdk/1.x/pdk.html) version 1.5.0 or higher ... `

**Comment by jpogran** [comment]
@michaeltlombardi @glennsarti Instead of doing two separate tickets, I changed the wording in this PR to support both adding the new PDK as source feature as well as documenting the Puppet-Agent staying as default. I didn't want to lose the wording of this PR in the new point release. Please review

File: README.md [hunk 2]
```diff
@@ -183,15 +193,92 @@ Settings:
...
+  "puppet.puppetAgentDir":"D:/programs/puppet"
+}
+```
 
 ## Experience a Problem?
 
 ### Puppet Agent Install
 
-A commonly encountered problem is not having Puppet Agent installed on the computer you are running VSCode on. As noted in the [Requirements section](https://github.com/lingua-pupuli/puppet-vscode/blob/master/README.md#requirements), you will need to have Puppet Agent installed in order to fully use this extension, as the extension uses the Puppet binaries and the Ruby language bundled into the agent install in order to function.
+A commonly encountered problem is not having the PDK or Puppet Agent installed on the computer you are running VSCode on. As noted in the [Requirements section](https://github.com/lingua-pupuli/puppet-vscode/blob/master/README.md#requirements), you will need to have the PDK or Puppet Agent installed in order to fully use this extension, as the extension uses the Puppet binaries and the Ruby language bundled into the pdk or agent install in order to function.
```
Comment (michaeltlombardi) [inline README.md]: `pdk` on last line should be capitalized.

File: README.md [hunk 3]
```diff
@@ -183,15 +193,92 @@ Settings:
...
 
-You can also install the extension without access to the internet byw following these [instructions](https://code.visualstudio.com/docs/editor/extension-gallery#_install-from-a-vsix.)
+You can also install the extension without access to the internet by following these [instructions](https://code.visualstudio.com/docs/editor/extension-gallery#_install-from-a-vsix.)
+
+## Configuration
+
+The Puppet VSCode extension can use the PDK or the Puppet Agent to provide advanced features, which can be set using two configuration options: `puppet.installType` and `puppet.puppetAgentDir`.
+
+- By default the `puppet.installType` is set to `agent`, and allows you to choose to use `pdk` or `agent`.
+
```
Comment (michaeltlombardi) [inline README.md]: This line can probably be removed since it's between two bullet points.

File: README.md (hunk 2 above)
Comment (jpogran) [reply README.md]: 👍

File: README.md (hunk 3 above)
Comment (jpogran) [reply README.md]: 👍

**Pull request merged**